### ISSUE TITLE
About the config file to train the model on nuScenes test set

### ISSUE THREAD

**Issue by zheng0819** [zhanggang001/HEDNet#23]
Hi, I am interested in your work, but one thing I noticed is that the profile of your safdnet model on nuscenes is not the same as what you have mentioned in the paper, for example, the fade strategy mentioned in the paper is in the last 5 epochs, while in the code it is in the last 4 epochs, so which one should I go with? Thanks!

**Comment by zhanggang001** [comment]
> Hi, I am interested in your work, but one thing I noticed is that the profile of your safdnet model on nuscenes is not the same as what you have mentioned in the paper, for example, the fade strategy mentioned in the paper is in the last 5 epochs, while in the code it is in the last 4 epochs, so which one should I go with? Thanks!

Sorry for late reply. Using 4 or 5 epochs in the fade strategy results in similar performance. For the nuScenes test, one important thing is to use both training and validation sets to train the model (also generate the gt_database using both training and validation sets), following the common setting.

**Comment by zhanggang001** [comment]
> Hi, I am interested in your work, but one thing I noticed is that the profile of your safdnet model on nuscenes is not the same as what you have mentioned in the paper, for example, the fade strategy mentioned in the paper is in the last 5 epochs, while in the code it is in the last 4 epochs, so which one should I go with? Thanks!

I check the paper again. Our original implementation is based on mmdet3d, where we used 5 epochs in the faded strategy. The released code is based on OpenPCDet, where we used 4 epochs in the faded strategy.  According to our experience, using 4 or 5 epochs in the faded strategy yields similar performance.

**Comment by zheng0819** [comment]
> > Hi, I am interested in your work, but one thing I noticed is that the profile of your safdnet model on nuscenes is not the same as what you have mentioned in the paper, for example, the fade strategy mentioned in the paper is in the last 5 epochs, while in the code it is in the last 4 epochs, so which one should I go with? Thanks!
> 
> Sorry for the late reply. Using 4 or 5 epochs in the faded strategy results in similar performance. For the nuScenes test, one important thing is to use both training and validation sets to train the model (also generate the gt_database using both training and validation sets), following the common setting.
> 
> I check the training log and paper again. In the faded strategy, we used 4 epochs when employing the OpenPCDet framework, and 5 epochs when employing the mmdetection3d framework to implement the TransFusion-L. We made a mistake in the paper. Sorry for the confusion.

Thanks for your reply, I have some more questions, for the configuration file, it is not only the faded strategy that is different from the paper, but also the voxel size ([0.075, 0.075, 0.2] in the paper, [0.1, 0.1, 0.2] in the code), the kernel size of group pooling for the SAFD-Net ( [9, 15, 5, 5] in the paper, [9, 15, 3, 3] in the code), num_proposals of the detection header (200 in the paper, 300 in the code), and the initial learning rate (0.003 in the paper, 0.001 in the code). Should all these settings prevail in the code or in the paper? Also, for the gt_database generation, is it only necessary to change 'train' in INFO_PATH of nuscenes_dataset.yaml to [nuscenes_infos_10sweeps_train.pkl , nuscenes_infos_10sweeps_val. pkl]? Thanks a lot!

**Comment by zhanggang001** [comment]
> > > Hi, I am interested in your work, but one thing I noticed is that the profile of your safdnet model on nuscenes is not the same as what you have mentioned in the paper, for example, the fade strategy mentioned in the paper is in the last 5 epochs, while in the code it is in the last 4 epochs, so which one should I go with? Thanks!
> > 
> > 
> > Sorry for the late reply. Using 4 or 5 epochs in the faded strategy results in similar performance. For the nuScenes test, one important thing is to use both training and validation sets to train the model (also generate the gt_database using both training and validation sets), following the common setting.
> > I check the training log and paper again. In the faded strategy, we used 4 epochs when employing the OpenPCDet framework, and 5 epochs when employing the mmdetection3d framework to implement the TransFusion-L. We made a mistake in the paper. Sorry for the confusion.
> 
> Thanks for your reply, I have some more questions, for the configuration file, it is not only the faded strategy that is different from the paper, but also the voxel size ([0.075, 0.075, 0.2] in the paper, [0.1, 0.1, 0.2] in the code), the kernel size of group pooling for the SAFD-Net ( [9, 15, 5, 5] in the paper, [9, 15, 3, 3] in the code), num_proposals of the detection header (200 in the paper, 300 in the code), and the initial learning rate (0.003 in the paper, 0.001 in the code). Should all these settings prevail in the code or in the paper? Also, for the gt_database generation, is it only necessary to change 'train' in INFO_PATH of nuscenes_dataset.yaml to [nuscenes_infos_10sweeps_train.pkl , nuscenes_infos_10sweeps_val. pkl]? Thanks a lot!

1.For model training, we set the num_proposals to 200. During model inference, we set the num_proposals to 200 for the validation set and 300 for the test set, following the settings in the original TransFusion-L paper.

2.We initially implemented HEDNet on the nuScenes dataset using MMDetection3D for better performance. To unify the codebase with OpenPCDet, we adopted a 2D sparse backbone for both SAFDNet and HEDNet on the nuScenes dataset (see the [issue](https://github.com/zhanggang001/HEDNet/issues/22)). Since we borrowed the experimental settings directly from the HEDNet paper, some mistakes were carried over. We will fix these in the paper. Please follow the settings in the released code to run your experiments.

3.To submit results to the test server, we trained the models on both the training and validation sets (and generated the gt_database from both sets as well). I recall that we merged the data annotations and did not change the configurations (this was done a long time ago, so I’m not entirely sure; my apologies).
